### ISSUE TITLE
[🐞]: Loading Minecraft Slower

### ISSUE THREAD

**Issue by FriedrichNietzsche8** [MehVahdJukaar/WoodGood#750]
### Code of Conduct

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue still persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [X] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version - Loader (DO NOT FORGET TO SELECT THE CORRECT VERSION)

1.21.1 - FABRIC

### EveryCompat Version

everycomp-1.21-2.8.4-fabric

### Moonlight Lib Version

moonlight-1.21-2.17.12-fabric

### Issue with mods

Judging by the logs it seems to be an issue with "everycomp:mcf/enchanted/" missing textures in model, but you should see for yourself

### Issue Detail

Takes an excruciatingly long time to load (about 5 minutes), takes normal amount of time load without Every Compat.
In version 2.8.2 it took normal amount of time, 2.8.3 made Minecraft crash at startup, but that was fixed in 2.8.4 (along with a crash in the server making department), although there's this issue now. In the log it start at [20:31:32]

</body>
</html>

### OPTIONAL: Latest.log | Crash-report Attachment

https://mclo.gs/MNLWiz4

**Comment by Xelbayria** [comment]
Has Macaw's Fences & Walls recently been updated? That's the only thing I can think of. 

EDIT: 
Just checked the curseforge, there is no recently update for Macaw's Fences & Walls. Have you allocated enough RAM to Minecraft? 

☝🏻 this is about 
> Takes an excruciatingly long time to load (about 5 minutes)

and for the missing textures from BWG and Enchanted. I think i know why. I'll need to include them in the module.

**Comment by FriedrichNietzsche8** [comment]
I had allocated exactly 4 GBs (4,608) of RAM in that moment,
now I have allocated 6 GBs (6,656) of RAM and the result is the same as 4 GBs if not even longer

**Comment by Xelbayria** [comment]
Can you try downgrade **Moonlight Lib** to v2.13.44 or older and check if that make a difference. There are reports from **Supplementaries** talking about how slow loading world have become with the v2.13.45 of **Moonlight Lib**

**Comment by Xelbayria** [comment]
Alright. the performance issue will take some times to investigate. The missing textures problem is easy to fix. 

EDIT:
which mod is the "enchanted"? I don't know which one?

**Comment by FriedrichNietzsche8** [comment]
https://modrinth.com/mod/enchanted-witchcraft
https://www.curseforge.com/minecraft/mc-mods/enchanted-witchcraft
Called "enchanted-fabric-1.21.1-4.0.7-beta.jar", this is the only mod I found that has Enchanted as name and is enabled

**Comment by Xelbayria** [comment]
oh it's Enchnated-Witchcraft. thank you. 

the texture problem with BWG (Oh The Biomes We've gone) for hedge is fixed.

**Comment by Xelbayria** [comment]
make that "Fixed 2 of 3 `#750`" 

Enchanted problem is fine now. I improved the code in Macaw's Fences & Walls to find the correct texture for leaves.

**Comment by FriedrichNietzsche8** [comment]
I tested with the disablement of Enchanted: Witchcraft just in case that was the culprit, still as slow
A theory I had (originating from Enchanted: Witchcraft not being on the compatibilities list) which might've just been proven to be wrong is that Enchanted: Witchcraft activated the part of the code that searches for compatibility dependencies because it's named similarly, causing it to run really slow and attempting to put textures in a model that isn't there,
However I think that'll just result in a crash than slowing REALLY down, and now that I have tested with it disabled it still slows down so might be just the texture errors

**Comment by Xelbayria** [comment]
Yeah. I ran test on the improved code in Macaw's Fences & Walls via Every Compat (the module) 

and i had no problem loading the world. it's quite normal. With BWG, and then with Enchanted. Something is definitely causing the slow performance. 

EDIT: 

> now that I have tested with it disabled it still slows down so might be just the texture errors

Actually, textures are generated during the booting/launching of minecraft. what get loaded during the "loading world" are: tags, recipes, and loot-tables

EDIT 2: 

Can i get the new latest.log with v2.8.4, i want to take a look to see if I can find the clue of what's causing the slow performance. 

Just to be clear, this is about your CLIENT, right?  not about you joining the server.

**Comment by FriedrichNietzsche8** [comment]
What I was about to say before the edit: "Yeah I have high suspicions that it's because of the texture errors. Oh also do note it's not an issue of loading the world, it's of the startup loading, i. e. launching Minecraft"
The log of the test without Enchanted: Witchcraft?
And yes it is about the client, the bot put it as server and I have no idea how to change it

**Comment by Xelbayria** [comment]
Ah i misunderstood you. so it was the time of startup loading. Alright, i'll let u know when the newer version is uploaded. 

> And yes it is about the client, the bot put it as server and I have no idea how to change it

that's because you used the keyword "server" in your first post

here this: `in the server making department), although there's this issue now. In the log it start at [20:31:32]`    
-------------------☝🏻 

EDIT: You could have let me know, so I could remove the wrong label.

**Comment by FriedrichNietzsche8** [comment]
Yeah I shall start every Issue Detail with CLIENT/SERVER ISSUE from now on if I have more to report
Here's the log that was requested, with Enchanted: Witchcraft disabled
https://mclo.gs/MmA9h6Y

**Comment by Xelbayria** [comment]
👍🏻 

When you start playing with the minecraft, get the ID of 2 woodTypes: for logs, planks and leaves
- mynethersdelight:powdery
- desolation:charred

Something should have changed in these 2 wood mods.

**Comment by FriedrichNietzsche8** [comment]
you want me to startup minecraft (with Every Compat 2.8.4 and Enchanted: Witchcraft), start a singleplayer world, and tell you which Logs, Planks and Leaves of the two mods have the IDs "mynethersdelight:powdery" and "desolation:charred", is that correct? (asking cuz not sure I understood completely)

**Comment by FriedrichNietzsche8** [comment]
Okay, a turn of events were all caused by this request.
I tried to create a world with Every Compat 2.8.4 (and Enchanted: Witchcraft) enabled and it took more than 20 minutes, which is where I pull the plug
Tried to do it with Every Compat disabled, same thing
Then I remembered I was using a client-side pack to test it, and I should enable the server-side... side of the mods.
1. Minecraft booted up at startup at a normal speed
2. Creating a Singleplayer World was normal
3. Here's IDs
---
My Nether's Delight Refabricated:
mynethersdelight:powdery_block
mynethersdelight:stripped_powdery_block
mynethersdelight:powdery_planks
mynethersdelight:powdery_stairs
mynethersdelight:powdery_slab
mynethersdelight:powdery_mosaic
mynethersdelight:powdery_mosaic_stairs
mynethersdelight:powdery_mosaic_slab
mynethersdelight:powdery_fence
mynethersdelight:powdery_fence_gate
mynethersdelight:powdery_door
mynethersdelight:powdery_trapdoor
mynethersdelight:powdery_pressure_plate
mynethersdelight:powdery_button
mynethersdelight:powdery_sign
mynethersdelight:powdery_hanging_sign
---
---
Desolation:
desolation:charred_boat
desolation:charred_chest_boat
desolation:charred_branches
desolation:charred_button
desolation:charred_door
desolation:charred_fence
desolation:charred_fence_gate
desolation:charred_hanging_sign
desolation:charred_log
desolation:charred_planks
desolation:charred_pressure_plate
desolation:charred_sapling
desolation:charred_sign
desolation:charred_slab
desolation:charred_stairs
desolation:charred_trapdoor
desolation:charred_wood
desolation:stripped_charred_log
desolation:stripped_charred_wood
---
There's also stuff like these but i'm not sure
mynethersdelight:powdery_cabinet
mynethersdelight:powdery_torch
desolation:charred_soil

**Comment by FriedrichNietzsche8** [comment]
But yeah the issue of minecraft booting really slow seems to have been fixed by enabling another mod

**Comment by Xelbayria** [comment]
oh Sorry I forgot to tell you, you don't need to use EveryCompat. just get the Ids of these woodTypes from 2 wood mods. 😓   20 minutes. wow. 

EDIT: 

i checked Moonlight Lib. these 2 woodTypes need to be updated. Thanks for the IDs.

**Comment by FriedrichNietzsche8** [comment]
Forgor to close this
